subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -137,10 +140,19 @@ private void configureCredentials(HttpClientBuilder builder, CredentialsProvider
             useCredentials(credentialsProvider, AuthScope.ANY_HOST, AuthScope.ANY_PORT, authentications);
 
             // Use preemptive authorisation if no other authorisation has been established
-            builder.addInterceptorFirst(new PreemptiveAuth(new BasicScheme(), isPreemptiveEnabled(authentications)));
+            builder.addInterceptorFirst(new PreemptiveAuth(getAuthScheme(authentications), isPreemptiveEnabled(authentications)));
```

Comment:
I don't think we need to change this because it's merely the fallback when the provided credentials have failed. Or am I missing something?

Reply:
It won't work without.

(I have a test project with a dependency installed to private gitlab.com repository which I use for end-to-end testing, and the header authentication is not send if this line is not changed.)

Reply:
Ok, looking closer, I now see why. Thanks!